### ISSUE TITLE
API method get_ns_ips returns incorrect result

### ISSUE THREAD

**Issue by aabdnn** [zonemaster/zonemaster-backend#163]
I sent this input to the backend:

```
{
"method": "get_ns_ips",
"params": "ripe.net"
}
```

I expected to get back the IP addresses of ripe.net's name servers. Instead, I got back the A record of ripe.net:

```
{"jsonrpc":"2.0","id":null,"result":[{"ripe.net":"193.0.6.139"},{"ripe.net":"2001:67c:2e8:22::c100:68b"}]}
```

**Comment by mtoma** [comment]
Hi Anand,

This method is used by the frontend GUI code to populate the list of IP addresses (in the undelegated domain name test tab) for a given NS name (typically one IPv4 and one IPv6 IP address).

So the right use case would be {"method": "get_ns_ips","params":"c2.authdns.ripe.net"}. 
Which will return {"jsonrpc":"2.0","id":null,"result":[{"c2.authdns.ripe.net":"162.159.25.153"},{"c2.authdns.ripe.net":"2400:cb00:2049:1::a29f:1999"}]}

Michal

----- Original Message -----

**Comment by aabdnn** [comment]
Okay, I understand now. Perhaps the method name could be clearer, eg. "get_addresses"

**Comment by mtoma** [comment]
Possibly, yes.
I'll keep the issue and see if it is possible to change the parameter.

----- Original Message -----

**Comment by mtoma** [comment]
Can be changed if we agree on a new method name.
TBD during F2F.

**Comment by sandoche2k** [comment]
"get_ns_ips" should have an alias to "get_host_by_name"

**Comment by mtoma** [comment]
Is the Validation Layer PR merged yet?

**Comment by matsduf** [comment]
@mtoma, no, see #399.

**Comment by sandoche2k** [comment]
@mtoma @matsduf close?

**Comment by matsduf** [comment]
Closed since it has been solved by PR #421
